lib/database/mongodb.ts

```diff
@@ -10,6 +10,7 @@ import { getReplacementFields } from "./util";
 export type PaginatedWordsRequest = {
   readonly page: number;
   readonly itemsPerPage: number;
+  readonly tags: ReadonlyArray<string> | undefined;
```

Comment:
Maybe `readonly tags?: ReadonlyArray<string>`? Or is it important that the field is present but can be `undefined`?

Reply:
I might be mistaken, but I think `tags?: T` is just syntax sugar for `tags: T | undefined`. I can change it to `tags?` for conciseness.

Reply:
Changed to `tags?`

Reply:
It is not the same actually, check this [TS playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAKhDOwoF4oG8BQUoA8BcUiATgJYB2A5lAD5QCuZAJhAGbkSMYC+GGoksBMABMKdFlwB+AsXIVuvAMYB7MoigALCABttygnHWo0PFWqQB3ZUW2MDQ0ca5A)